Marlin/src/HAL/LPC1768/fast_pwm.cpp

```diff
@@ -25,7 +25,9 @@
 #include <pwm.h>
 
 void set_pwm_duty(const pin_t pin, const uint16_t v, const uint16_t v_size/*=255*/, const bool invert/*=false*/) {
-  LPC176x::pwm_write_ratio(pin, invert ? 1.0f - (float)v / v_size : (float)v / v_size);
+  if (!LPC176x::pin_is_valid(pin)) return;
+  if (LPC176x::pwm_attach_pin(pin)) 
```

Comment:
Would it make more sense to call `LPC176x::pwm_attach_pin` in `set_pwm_frequency` instead of (or in addition to) in `set_pwm_duty` since (presumably) `set_pwm_frequency` is called at initialization, while `set_pwm_duty` is more likely to be called as part of the operation loop?

Reply:
Yes it could be a better option and I will test that possibility as well.